packages/kinvey-datastore2/test/datastore.spec.js

```diff
@@ -0,0 +1,79 @@
+import { expect } from 'chai';
+import { spy } from 'sinon';
+import { DataStore, DataStoreType, collection } from '../src';
+
+describe('DataStoreType', () => {
+  it('should have type Cache', () => {
+    expect(DataStoreType.Cache).to.not.be.undefined;
+  });
+
+  it('should have type Network', () => {
+    expect(DataStoreType.Network).to.not.be.undefined;
+  });
+
+  it('should have type Sync', () => {
+    expect(DataStoreType.Sync).to.not.be.undefined;
+  });
+});
+
+describe('DataStore', () => {
+  describe('constructor', () => {
+    it('should throw and error if a collection is not defined', () => {
+      expect(() => new DataStore()).to.throw();
+    });
+
+    it('should set the collection', () => {
+      const collectionName = 'test';
+      const datastore = new DataStore(collectionName);
+      expect(datastore.collection).to.equal(collectionName);
+    });
+  });
+
+  describe('save()', () => {
+    let datastore;
+
+    before(() => {
```

Comment:
Why not just do `const dataStore = new DataStore()`, do we need the before?

Reply:
I was trying to stick to Mocha conventions for setup and teardown.

Reply:
Okay, I'm not too familiar with those. If that's the way the say we should do it, I'm fine with that.